src/core/border_router/routing_manager.hpp

```diff
@@ -624,10 +625,18 @@ class RoutingManager : public InstanceLocator
         void HandleRouterTimer(void);
 
     private:
+#if !OPENTHREAD_CONFIG_BORDER_ROUTING_USE_HEAP_ENABLE
         static constexpr uint16_t kMaxRouters = OPENTHREAD_CONFIG_BORDER_ROUTING_MAX_DISCOVERED_ROUTERS;
         static constexpr uint16_t kMaxEntries = OPENTHREAD_CONFIG_BORDER_ROUTING_MAX_DISCOVERED_PREFIXES;
+#endif
 
-        class Entry : public LinkedListEntry<Entry>, public Unequatable<Entry>, private Clearable<Entry>
+        class Entry : public LinkedListEntry<Entry>,
+                      public Unequatable<Entry>,
+                      private Clearable<Entry>
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_USE_HEAP_ENABLE
+            ,
+                      public Heap::Allocatable<Entry>
+#endif
```

Comment:
Purely for code style, could we move this above `Clearable` to avoid the comma on its own line?

Reply:
Good idea. `clang-format` does not make this look good.